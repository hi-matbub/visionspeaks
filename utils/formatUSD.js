/* eslint-disable import/prefer-default-export */
const formatUSD = (number) => {
  // for inconsistent types
  const string = JSON.stringify(parseFloat(number));
  const { length } = string || number;
  const str = `${string.substring(0, length - 2)}.${string.substring(
    length - 2
  )}`;
  return parseFloat(str).toFixed(2);
};

export { formatUSD };
