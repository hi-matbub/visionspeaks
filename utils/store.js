exports.store = {
  name: 'Dog-Gone Goods',
  logo: '/logo.svg',
  hero: '/hero.svg',
  url: '',
  description: 'A dog treat shop',
  about: 'Everyone loves a happy dog!',
  contact: 'youremail@here.com',
  social: {
    facebook: '',
    twitter: '',
    instagram: '',
  },
};
