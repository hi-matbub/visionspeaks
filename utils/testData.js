// used for testing purposes

exports.testData = {
  store: {
    name: 'Dog-Gone Goods',
    logo: '/logo.svg',
    hero: '/hero.svg',
    url: '',
    description: 'A dog treat shop',
    about: 'Everyone loves a happy dog!',
    contact: 'youremail@here.com',
    social: { facebook: '', twitter: '', instagram: '' },
  },
  products: [
    {
      id: 'prod_IRbpyL6fgse3n',
      object: 'product',
      active: true,
      attributes: [],
      created: 1606151601,
      description: 'demo product description',
      images: ['img.jpg'],
      livemode: false,
      metadata: {
        price: 287,
      },
      name: 'Large Demo Product',
      statement_descriptor: null,
      type: 'service',
      unit_label: null,
      updated: 1606153202,
    },
    {
      id: 'prod_IRbouvicOUom16',
      object: 'product',
      active: true,
      attributes: [],
      created: 1606151551,
      description: 'demo product description',
      images: ['img.jpg'],
      livemode: false,
      metadata: {
        price: 287,
      },
      name: 'Demo Product',
      statement_descriptor: null,
      type: 'service',
      unit_label: null,
      updated: 1606153221,
    },
  ],
};
