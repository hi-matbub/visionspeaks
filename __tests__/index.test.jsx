import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Index from '../pages/index';
import { testData } from '../utils/testData';

it('index renders smoke test without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Index data={testData} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('index renders unchanged', () => {
  const tree = renderer.create(<Index data={testData} />).toJSON();
  expect(tree).toMatchSnapshot();
});
