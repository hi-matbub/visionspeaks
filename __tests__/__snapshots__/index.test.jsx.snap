// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index renders unchanged 1`] = `
<div
  className="content"
>
  <div
    className="d-flex justify-content-between w-100 fixed-top"
    style={
      Object {
        "background": "#FFFFFF !important",
      }
    }
  >
    <nav
      className="justify-content-between fixed-top navbar navbar-expand-lg navbar-light fixed-top"
      style={
        Object {
          "backgroundColor": "#FFFFFF !important",
        }
      }
    >
      <img
        alt="React Bootstrap logo"
        className="d-inline-block align-top"
        height="75"
        src="/logo.svg"
      />
      <a
        className="navbar-brand"
        href="/"
      >
        Dog-Gone Goods
      </a>
      <div
        className="navbar-nav"
        onKeyDown={[Function]}
      >
        <button
          onClick={[Function]}
          style={
            Object {
              "background": "none",
              "border": "0",
            }
          }
          type="button"
        >
          <div
            className="bg-warning scale-in-center"
            style={
              Object {
                "borderRadius": "50%",
                "height": "15px",
                "position": "absolute",
                "right": "14px",
                "top": "28px",
                "width": "15px",
              }
            }
          />
          <img
            alt="Checkout"
            src="/cart.png"
            style={
              Object {
                "cursor": "pointer",
                "height": "20px",
              }
            }
          />
        </button>
      </div>
    </nav>
  </div>
  <main>
    <div
      className="hero_"
      style={
        Object {
          "backgroundImage": "url(/hero.svg)",
          "backgroundPosition": "bottom",
          "backgroundRepeat": "no-repeat",
          "backgroundSize": "cover",
        }
      }
    />
    <div
      className="container"
    >
      <div
        className="align-items-center justify-content-center row"
      >
        <div
          className="col-lg-8 col-md-9 col-sm-12"
        >
          <h2
            className="my-5"
          >
            Products
          </h2>
        </div>
      </div>
      <div
        className="align-items-center justify-content-center mb-2 align-items-center justify-content-center row"
      >
        <div
          className="col-lg-8 col-md-9 col-sm-12"
        >
          <button
            className="btn btn-outline-dark  d-flex w-100"
            onClick={[Function]}
            style={
              Object {
                "borderRadius": "6px",
                "cursor": "pointer",
              }
            }
            type="button"
          >
            <img
              alt="Large Demo Product"
              className="product_img_prc"
              src="img.jpg"
            />
            <div
              className="d-flex w-100 flex-column align-items-end justify-content-center text-right"
            >
              <h5
                className="mt-3 mb-2 ellipse"
              >
                Large Demo Product
              </h5>
              <span
                className="text-muted mb-3"
              >
                $
                2.87
                 USD
              </span>
            </div>
          </button>
        </div>
      </div>
      <div
        className="align-items-center justify-content-center mb-2 align-items-center justify-content-center row"
      >
        <div
          className="col-lg-8 col-md-9 col-sm-12"
        >
          <button
            className="btn btn-outline-dark  d-flex w-100"
            onClick={[Function]}
            style={
              Object {
                "borderRadius": "6px",
                "cursor": "pointer",
              }
            }
            type="button"
          >
            <img
              alt="Demo Product"
              className="product_img_prc"
              src="img.jpg"
            />
            <div
              className="d-flex w-100 flex-column align-items-end justify-content-center text-right"
            >
              <h5
                className="mt-3 mb-2 ellipse"
              >
                Demo Product
              </h5>
              <span
                className="text-muted mb-3"
              >
                $
                2.87
                 USD
              </span>
            </div>
          </button>
        </div>
      </div>
    </div>
  </main>
  <footer>
    <div
      className="my-5 container"
    >
      <div
        className="align-items-center justify-content-center row"
      >
        <div
          className="col-lg-8 col-md-9 col-sm-12"
        >
          <h2>
            More Info
          </h2>
          <div
            className="d-flex flex-wrap"
          >
            <div
              className="d-flex flex-column"
            >
              <button
                className="btn btn-outline-dark mt-2 pl-1 text-left"
                id="About"
                onClick={[Function]}
                style={
                  Object {
                    "border": "0",
                    "padding": "0",
                    "width": "200px",
                  }
                }
                type="button"
              >
                About
              </button>
              <button
                className="btn btn-outline-dark mt-2 pl-1 text-left"
                id="FAQ"
                onClick={[Function]}
                style={
                  Object {
                    "border": "0",
                    "padding": "0",
                    "width": "200px",
                  }
                }
                type="button"
              >
                FAQ
              </button>
              <button
                className="btn btn-outline-dark mt-2 pl-1 text-left"
                id="Contact"
                onClick={[Function]}
                style={
                  Object {
                    "border": "0",
                    "padding": "0",
                    "width": "200px",
                  }
                }
                type="button"
              >
                Contact
              </button>
            </div>
            <div
              className="d-flex flex-column"
            >
              <button
                className="btn btn-outline-dark mt-2 pl-1 text-left"
                id="Terms and Conditions"
                onClick={[Function]}
                style={
                  Object {
                    "border": "0",
                    "padding": "0",
                    "width": "200px",
                  }
                }
                type="button"
              >
                Terms and Conditions
              </button>
              <button
                className="btn btn-outline-dark mt-2 pl-1 text-left"
                id="Privacy Policy"
                onClick={[Function]}
                style={
                  Object {
                    "border": "0",
                    "padding": "0",
                    "width": "200px",
                  }
                }
                type="button"
              >
                Privacy Policy
              </button>
              <button
                className="btn btn-outline-dark mt-2 pl-1 text-left"
                id="Shipping and Returns"
                onClick={[Function]}
                style={
                  Object {
                    "border": "0",
                    "padding": "0",
                    "width": "200px",
                  }
                }
                type="button"
              >
                Shipping and Returns
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="d-flex justify-content-center col"
        >
          <p
            className="m-5 text-center"
          >
            Copyright 
            <br />
             
            2020
          </p>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
